control/state.py

```
@@ -757,38 +762,63 @@ def namespace_only_lb_group_id_changed(self, old_val, new_val):
             return (False, None)
         return (True, new_req.anagrpid)
 
-    def host_only_keys_changed(self, old_val, new_val):
+    def host_only_key_changed(self, old_val, new_val):
+        # If only the dhchap key has changed we can use change_key request instead of re-adding the host
         old_req = None
         new_req = None
         try:
             old_req = json_format.Parse(old_val, pb2.add_host_req(), ignore_unknown_fields=True )
         except Exception as ex:
```

@gbregman looks awesome! could we catch ParseError (from google.protobuf.json_format) see [here](https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html#google.protobuf.json_format.ParseError)? in `control/state.py`

---

I used the same code we used before, where we catch all exceptions, not just ParseError. Why do we need to only catch that?